Frameworks/UIKit/StarboardXaml/LayerCoordinator.cpp

```diff
@@ -752,10 +756,15 @@ void LayerCoordinator::_ApplyContentCenter(FrameworkElement^ element, Rect conte
     if (contentCenter.Equals(Rect(0, 0, 1.0, 1.0))) {
         image->NineGrid = Thickness(0, 0, 0, 0);
     } else {
-        int left = static_cast<int>(contentCenter.X * contentSize.Width);
-        int top = static_cast<int>(contentCenter.Y * contentSize.Height);
-        int right = (static_cast<int>(contentSize.Width) - (left + (static_cast<int>(contentCenter.Width * contentSize.Width))));
-        int bottom = (static_cast<int>(contentSize.Height) - (top + (static_cast<int>(contentCenter.Height * contentSize.Height))));
+        // When creating the NineGrid, we need to take the scaled image dimensions into account
+        const double contentScale = _GetContentScale(element);
+        const double imageWidth = contentSize.Width * contentScale;
```

Comment:
I think I prefer this approach.  This is the only place we have to 'undo' the scale, whereas the contentSize is used in more places and it was more convenient to divide the scale before setting the contentSize.  For example, the _ApplyContentGravity function would be a bit messier.  Thoughts?

---
In reply to: [92278738](https://github.com/Microsoft/WinObjC/pull/1530#discussion_r92278738) [](ancestors = 92278738)